reporting-service/src/main/java/it/gov/pagopa/reporting/service/GPDService.java

```diff
@@ -24,12 +34,52 @@ public static GPDService getInstance() {
         return instance;
     }
 
-    public boolean setReport(String idPA, PaymentOption paymentOption) {
-        Response response = ClientBuilder.newClient()
-                .target(gpdHost + String.format(gpdpaymentOptionsService, idPA, paymentOption.getOptionId(), paymentOption.getTransferId()))
-                .request()
-                .accept(MediaType.APPLICATION_JSON)
-                .post(Entity.text(""));
-        return response.getStatus() == 200;
+    /**
+     * set report with retry
+     *
+     * @param idPA          fiscal code
+     * @param paymentOption payment option
+     * @param logger        for logging
+     * @return return true if the report is set correctly
+     */
+    @SneakyThrows
+    public boolean setReport(String idPA, PaymentOption paymentOption, Logger logger) {
+        int maxAttemptsValue = Integer.parseInt(Optional.ofNullable(maxAttemps)
+                .orElse("1")); // for testing
+        int delayValue = Integer.parseInt(Optional.ofNullable(delay)
+                .orElse("0")); // for testing
+        for (int i = 1; i <= maxAttemptsValue; i++) {
```

Comment:
enqueue newly the message
As made for https://github.com/pagopa/pagopa-debt-position/blob/43dff4fc8134f574a68028d75213cbf4c8d657a7/reporting-batch/src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java#L72